core/src/test/java/com/datastrato/gravitino/lock/TestTreeLock.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.lock;
+
+import static com.datastrato.gravitino.lock.TestLockManager.getConfig;
+
+import org.junit.jupiter.api.Test;
+
+class TestTreeLock {
+
+  @Test
+  void testLock() {
+    LockManager lockManager = new LockManager(getConfig());
+
+    for (int i = 0; i < 1000; i++) {
+      TreeLock lock = lockManager.createTreeLock(TestLockManager.randomNameIdentifier());
+      lock.lock(i % 2 == 0 ? LockType.READ : LockType.WRITE);
+      try {
+        // Business logic here.
+      } finally {
+        lock.unlock();
+      }
+    }
+  }
+}
```

Comment:
Add exception error tester ,  and check the lock refcount. 
```
 TreeLock lock = lockManager.createTreeLock(TestLockManager.randomNameIdentifier());
      lock.lock(i % 2 == 0 ? LockType.READ : LockType.WRITE);
      try {
        throw e
      } finally {
        lock.unlock();
      }

```

```
 TreeLock lock = lockManager.createTreeLock(TestLockManager.randomNameIdentifier());
     throw e
      lock.lock(i % 2 == 0 ? LockType.READ : LockType.WRITE);
      try {
        // Business logic here.
      } finally {
        lock.unlock();
      }
```

Reply:
added